subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -0,0 +1,716 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdio.h>
+#include <bluetooth/mesh/models.h>
+#include <sys/byteorder.h>
+#include "model_utils.h"
+#include "mesh/net.h"
+#include "mesh/access.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_MODEL)
+#define LOG_MODULE_NAME bt_mesh_scene_srv
+#include "common/log.h"
+
+#define SCENE_PAGE_SIZE SETTINGS_MAX_VAL_LEN
+
+struct __packed scene_data {
+	uint8_t len;
+	uint8_t elem_idx;
+	uint16_t id;
+	uint8_t data[];
+};
+
+static char *scene_path(char buf[9], uint16_t scene, bool vnd, uint8_t page)
+{
+	sprintf(buf, "%x/%c%x", scene, vnd ? 'v' : 's', page);
```

Comment:
Maybe this is common for this codebase, but this could actually be converted into 2 macro definitions `PATH_LEN` and `PATH_FORMAT`.

Reply:
The Zephyr and NCS Bluetooth subsystems generally don't create defines for things that only happen once, unless it's very unclear what is happening or if the contents of the define deserves an abstraction. 

Not adding defines for everything allows us to localize information. If I add a define here, I'll have to scroll to the top of the file to verify that this is working correctly. It also wouldn't really add any information that isn't obvious from the context. The same goes for "magic numbers", although they are arguably too heavily used throughout the stack.